sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/WebExceptionRetryPolicyE2ETests.java

```diff
@@ -0,0 +1,303 @@
+            serverGoneRule.disable();
+        }
+    }
+
+    @Test(groups = {"multi-master"}, dataProvider = "operationTypeProvider", timeOut = 8 * TIMEOUT)
+    public void dataPlaneRequestHttpTimeout(
+        FaultInjectionOperationType faultInjectionOperationType,
+        OperationType operationType) {
+
+        if (BridgeInternal
+            .getContextClient(this.cosmosAsyncClient)
+            .getConnectionPolicy()
+            .getConnectionMode() != ConnectionMode.GATEWAY) {
+            throw new SkipException("queryPlanHttpTimeoutWillNotMarkRegionUnavailable() is only meant for GATEWAY mode");
+        }
+
+        TestItem newItem = TestItem.createNewItem();
+        this.cosmosAsyncContainer.createItem(newItem).block();
+        FaultInjectionRule requestHttpTimeoutRule = new FaultInjectionRuleBuilder("requestHttpTimeoutRule" + UUID.randomUUID())
+            .condition(
+                new FaultInjectionConditionBuilder()
+                    .operationType(faultInjectionOperationType)
+                    .connectionType(FaultInjectionConnectionType.GATEWAY)
+                    .build())
+            .result(
+                FaultInjectionResultBuilders.getResultBuilder(FaultInjectionServerErrorType.RESPONSE_DELAY)
+                    .delay(Duration.ofSeconds(66))
+                    .times(4) // make sure it will exhaust the local region retry
+                    .build()
+            )
+            .build();
+
+        CosmosFaultInjectionHelper.configureFaultInjectionRules(this.cosmosAsyncContainer, Arrays.asList(requestHttpTimeoutRule)).block();
+
+
+        try {
+            CosmosDiagnostics cosmosDiagnostics =
+                this.performDocumentOperation(cosmosAsyncContainer, operationType, newItem).block();
+            System.out.println("dataPlaneRequestHttpTimeout() Diagnostics " + " " + cosmosDiagnostics);
+            validateDataPlaneRetryPolicyResponseTimeouts(cosmosDiagnostics);
+        } catch (Exception e) {
+            fail("dataPlaneRequestHttpTimeout() should succeed for operationType " + operationType, e);
+        } finally {
+            requestHttpTimeoutRule.disable();
+        }
+    }
+
+    private void validateDataPlaneRetryPolicyResponseTimeouts(CosmosDiagnostics cosmosDiagnostics) {
+        List<ClientSideRequestStatistics.GatewayStatistics> gatewayStatisticsList = diagnosticsAccessor.getClientSideRequestStatistics(cosmosDiagnostics)
+            .stream()
+            .map(ClientSideRequestStatistics::getGatewayStatisticsList)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
+
+        for (ClientSideRequestStatistics.GatewayStatistics gs : gatewayStatisticsList) {
+            for (RequestTimeline.Event event : gs.getRequestTimeline()) {
+                Duration durationInMillis = event.getDuration();
+                if (durationInMillis != null) {
+                    assertThat(durationInMillis.getSeconds()).isLessThanOrEqualTo(62);
```

Comment:
also worth adding the low bound comparison , like` duration.getSeconds > timeout based on the httpTimeoutPolicy`.  By just comparing the high bound, it will not catch the issue

Reply:
Added the lower bound check as well.